### PR TITLE
fix(security): filter git auth information

### PR DESCRIPTION
**Pull request by zhujian7**
**What this PR does / why we need it**:

Add your description

**Which issue(s) this PR is related to** *(optional, link to 3rd issue(s))*:

Fixes #

Reference to #

**Special notes for your reviewer**:

/cc @hyy0322 @supereagle @bbbmj @qianlei90 @muzi502 

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: build/resolver/git/entrypoint.sh [hunk 1]
```diff
@@ -223,6 +223,10 @@ pull() {
         fi
     fi
 
+    # filter sensitive information
+    git remote remove origin
+    git remote add origin "${NO_AUTH_SCM_URL}"
```
Comment (qianlei90) [inline build/resolver/git/entrypoint.sh:228]: 这两个命令可以用 `git remote set-url origin ${NO_AUTH_SCM_URL}` 代替吗？

**Comment by hyy0322** [comment]
LGTM

File: build/resolver/git/entrypoint.sh [hunk 2]
```diff
@@ -223,7 +223,10 @@ pull() {
         fi
     fi
 
-    cd $WORKDIR/data
+    # filter sensitive information
```
Comment (bbbmj) [inline build/resolver/git/entrypoint.sh:226]: what's the difference with this command? Can you add more annotation to explain what it changed and why it comes out?

Comment (zhujian7) [reply build/resolver/git/entrypoint.sh]: PTAL

File: build/resolver/git/entrypoint.sh [hunk 3]
```diff
@@ -223,7 +223,11 @@ pull() {
         fi
     fi
 
-    cd $WORKDIR/data
+    # filter sensitive information, otherwise users executing commands like 'git remote get-url origin'
```
Comment (bbbmj) [inline build/resolver/git/entrypoint.sh:226]: /lgtm

**Pull request merged**